crates/proof-impl/evm-ee-stf/src/primitives.rs

```diff
@@ -0,0 +1,78 @@
+use std::collections::HashMap;
+
+use alloy_consensus::{serde_bincode_compat, Header};
+use reth_primitives::{
+    revm_primitives::alloy_primitives::{Address, Bytes, FixedBytes, B256},
+    TransactionSignedNoHash, Withdrawal,
+};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use strata_reth_primitives::WithdrawalIntent;
+use strata_state::block::ExecSegment;
+
+use crate::mpt::{MptNode, StorageEntry};
+
+/// Public Parameters that proof asserts
+pub type EvmEeProofOutput = Vec<ExecSegment>;
+
+/// Public Parameters that proof asserts
+pub type EvmEeProofInput = Vec<EvmBlockStfInput>;
+
+/// Result of the block execution
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct EvmBlockStfOutput {
+    pub block_idx: u64,
+    pub prev_blockhash: FixedBytes<32>,
+    pub new_blockhash: FixedBytes<32>,
+    pub new_state_root: FixedBytes<32>,
+    pub txn_root: FixedBytes<32>,
+    pub withdrawal_intents: Vec<WithdrawalIntent>,
+    pub deposits_txns_root: FixedBytes<32>,
+}
+
+/// Necessary information to prove the execution of a Evm block.
+#[serde_as]
+#[derive(Debug, Default, Clone, PartialEq, Serialize, Deserialize)]
+pub struct EvmBlockStfInput {
+    /// The Keccak 256-bit hash of the parent block's header, in its entirety.
+    /// N.B. The reason serde_bincode_compat is necessary:
+    /// `[serde_bincode_compat]`(alloy_consensus::serde_bincode_compat)
+    #[serde_as(as = "serde_bincode_compat::Header")]
+    pub parent_header: Header,
+
+    /// The 160-bit address to which all fees collected from the successful mining of this block
+    /// be transferred.
+    pub beneficiary: Address,
+
+    /// A scalar value equal to the current limit of gas expenditure per block.
+    pub gas_limit: u64,
+
+    /// A scalar value equal to the reasonable output of Unix's time() at this block's inception.
+    pub timestamp: u64,
+
+    /// An arbitrary byte array containing data relevant to this block. This must be 32 bytes or
+    /// fewer.
+    pub extra_data: Bytes,
+
+    /// A 256-bit hash which, combined with the nonce, proves that a sufficient amount of
+    /// computation has been carried out on this block.
+    pub mix_hash: B256,
+
+    /// The state trie of the parent block.
+    pub parent_state_trie: MptNode,
+
+    /// The storage of the parent block.
+    pub parent_storage: HashMap<Address, StorageEntry>,
+
```

Comment:
I think `parent_storage` is a poor name for this, since it's really about the storage entries from the pre-state that we know we're going to touch.  Right?  Am I understanding this correctly?  It being from the parent block is secondary to that use.

Reply:
Yes, its the storage entries from the parent block which we are to touch.

Reply:
The point is that it makes it sound like it's the storage entries that *were* touched in the parent block's execution.  Not that it's the storage entries (in the pre-state) that *we're going to* touch.

Reply:
Did some refactoring around this and updated the comment.
But, I could not think of better naming here. 
let me know if you can suggest better choice of word here

Reply:
Yeah `pre_state_` is the most correct.